### ISSUE TITLE
Compatibility issue with Node.js 23.0.0: TypeError in @nodelib/fs.scandir

### ISSUE THREAD

**Issue by demonicattack** [nodelib/nodelib#110]
I encountered a compatibility issue with @nodelib/fs.scandir when using Node.js version 23.0.0. The error occurs due to an attempt to set a property on a Dirent object, which has only a getter.

![22](https://github.com/user-attachments/assets/6474aa58-f76b-4712-a9e1-1b058786b75b)

Ubuntu 24.04
Node.js 23.0.0
@nodelib/fs.scandir 3.0.0

**Comment by paschun** [comment]
Looks like it'll be resolved next release

https://github.com/nodejs/node/issues/55538

**Comment by paschun** [comment]
It throws on bun too though.
Seems modifying fs.Dirent is deprecated 
Can be patched with
```ts
Object.defineProperty(DirentFromStats.prototype, name, { get(){ return this[kStats][name]() }})
```

**Comment by mrmlnc** [comment]
This issue was fixed in the node@23.2 (https://github.com/nodejs/node/issues/55538).
